### DISCUSSION TITLE
v1.2.0

### DISCUSSION THREAD

**Discussion by OrangeJedi** [OrangeJedi/Aerial#135] [Announcements]
![image](https://user-images.githubusercontent.com/25063853/224231865-f8716c9a-ff9a-4c2b-bf18-21dc1bd4d504.png)

This update brings lots of new feature to Aerial including: multi-line text, video transitions, random text location, and a new icon. Enjoy!

# New Feature Summary 
* Multi-line text (https://github.com/OrangeJedi/Aerial/issues/26)
    * Text can now be displayed on up to 4 lines
    * Customize each line's type and font
* Multi-monitor text (https://github.com/OrangeJedi/Aerial/issues/112)
    * Choose which monitor a line can be displayed on
* Random/moving text locations (https://github.com/OrangeJedi/Aerial/issues/124)
* Video Transitions
    * Dissolve, fade, wipe, circle, and more!
    * Customize the speed
    * Randomize the type and direction of the transition
* Improvements to starting and closing Aerial
    * A keyboard shortcut to start the screensaver (https://github.com/OrangeJedi/Aerial/issues/71)
    * Aerial will not run when the computer is sleeping or locked
    * Aerial will automatically close itself when your computer locks or unlocks
    * You can now set a Aerial not lock your computer if you wake it up immediacy after it starts
    * You can now configure Aerial to not run on battery power
* New Icon (congratulations to @TheFreshPrins for wining our vote!)
* New display category: images
* Stretch or crop option for special screen sizes (https://github.com/OrangeJedi/Aerial/issues/119)
* Font selector now shows all the fonts when empty (https://github.com/OrangeJedi/Aerial/issues/45)
* Fixed two issues where a popup was showing on startup for some users (https://github.com/OrangeJedi/Aerial/issues/131 and https://github.com/OrangeJedi/Aerial/issues/132)
* Increased video loading performance
* You can now customize the video skip key
* Added a button to access the config files
* Smoothed out the config menu
* Updated dependencies

## How to Install
Download `Aerial Setup 1.2.0.exe` (or download the .zip and extract it) and run it. Aerial takes care of the rest. A configuration window will open where you can adjust the settings.

<hr /><em>This discussion was created from the release <a href='https://github.com/OrangeJedi/Aerial/releases/tag/v1.2.0'>v1.2.0</a>.</em>

**Comment by pparado** [comment]
I downloaded the update, and I was fiddling with the text. I clicked on update video cache, and it suddenly showed 66.xxGB? (it almost doubled from the former 35.6GB?

So, I first tried running the app using the shortcut keys. It worked... 
But I didn't quite get the customization I wanted for the text and their position. So, I kept going back to customizing.

When I couldn't figure out customizing the text and the position...I uninstalled v1.2.0

Then...this is where my troubles began. There is no more icon in the system tray.
So, I uninstalled any instance of Aerial, and reinstalled v1.1.0, so that I can get back to where I was before the update.
Unfortunately, there is no more icon in system tray and I cannot configure the app anymore.

So, I uninstalled any instance for the time being and would like to hear what others say.

**Reply by OrangeJedi** [reply]
I'm sorry to hear that you are having issues. It sounds like your config file got corrupted or accidentally misconfigured. You can try using the `Reset Aerial` button in whichever version you are running and see if that fixes it. You also may need to check the `Auto launch from system tray` option under the advanced menu. Keep in mind that they system tray icon changed in v1.2.0 and that Aerial needs to be installed and running to show in the tray.

Also, what problems were you running into with customizing the text that lead you do try installing v1.1.0? Perhaps the UI isn't as intuitive as I though and I need to write up some documentation.

**Reply by pparado** [reply]
I have removed all files, including the 'aerial' folder in C. 
I reverted to the older one, where .scr is placed in the C: 'Windows', and the option is in the 'change screensaver' dialog box.
I can confirm that the system tray icon has changed with v1.2.0, and it was running in the tray
The issue I encountered was: I can't properly place the text where I wanted them. They were showing up all over the place. Text that I placed on screen1, was showing up on screen 2 (and vice versa).  Text was showing up in the middle of 2 screens.  I was positioning them at the edges/corners but when they showed up somewhere else, I thought: I'll wait for others to share their experience.
This is what made me want to revert to v1.1.0

BTW, what is the screen width for? I was using 33% and it was ok.

Maybe a clean install will work?
Can I do a clean install of just v1.2.0?
Do I need to install v1.1.0 before upgrading to v1.2.0?

**Comment by pparado** [comment]
I did what I mentioned above, and it worked....for a while.

Now the video freezes after some time (I do not wait for it to happen)

There is a window popup with "javascript error in the main process"
C:\Users\User\AppData\Local\Programs\aerial\resources\app.asar\node_modules

So, the next time the app executes, it will not start.
Unless, I restart the app.
And then the whole scenario starts all over again.



![aerial java error](https://user-images.githubusercontent.com/118233508/224638093-6e0c8790-8ed8-4650-a9f2-4366f0218a12.jpg)

**Reply by OrangeJedi** [reply]
This is starting to look like a permissions issue with config file. I would review the `C:\Users\User\AppData\Roaming\aerial` directory and make sure that your logged in user has permission to write to the folder.

**Reply by ccunning70** [reply]
I get the same issue daily. Everyday it crashes with the same JavaScript error window that the previous user sent in. I checked my user permissions to the AppData folder you mentioned and I have full permissions to that folder.

**Comment by TheFreshPrins** [comment]
Nice to see a new version! And thanks for the winning icon!
